### PR TITLE
Fix uncaught throws and add retry for setattr. NFC

### PR DESCRIPTION
**Pull request by rstz**

### CODE REVIEW

File: pthreadfs/library_pthreadfs.js [hunk 1]
```diff
@@ -3119,25 +3121,46 @@ mergeInto(LibraryManager.library, {
...
+              else {
+                console.log(`FSAFS warning: Truncating an access handle failed with unexpected error ${e.name}. Trying again.`);
               }
-            } catch (e) {
-              if (!('code' in e)) throw e;
-              throw new PThreadFS.ErrnoError(-e.errno);
             }
+            await timeout(waiting_time_ms);
           }
+          console.log(`FSAFS warning: Truncating an access handle failed, aborting`);
```
Comment (fivedots) [inline pthreadfs/library_pthreadfs.js:3161]: Consider adding a note of the retries: "Truncating access handle failed after multiple attempts, aborting"

File: pthreadfs/library_pthreadfs.js [hunk 2]
```diff
@@ -3119,25 +3121,46 @@ mergeInto(LibraryManager.library, {
...
+            return;
           }
-          else {  // !ENVIRONMENT_IS_WEB
-            let useOpen = false;
-            let fileHandle = node.handle;
+          if (node.handle) {
+            await node.handle.truncate(attr.size);
+            return;
+          }
+          // On a worker without an open access handle, try two times to open an access handle.
```
Comment (fivedots) [inline pthreadfs/library_pthreadfs.js:3130]: s/two/three/

File: pthreadfs/library_pthreadfs.js [hunk 3]
```diff
@@ -3119,25 +3121,46 @@ mergeInto(LibraryManager.library, {
...
+                console.log(`FSAFS warning: Truncating an access handle failed with unexpected error ${e.name}. Trying again.`);
               }
-            } catch (e) {
-              if (!('code' in e)) throw e;
-              throw new PThreadFS.ErrnoError(-e.errno);
             }
+            await timeout(waiting_time_ms);
           }
+          console.log(`FSAFS warning: Truncating an access handle failed, aborting`);
+          // Opening an access handle was unsuccessful
```
Comment (fivedots) [inline pthreadfs/library_pthreadfs.js:3162]: Nit.: This comment feels redundant with the log above and the fact that it's a throw.

**Review by fivedots — APPROVED** [review]
LGTM!

File: pthreadfs/library_pthreadfs.js (hunk 1 above)
Comment (rstz) [reply pthreadfs/library_pthreadfs.js:3161]: Done.

File: pthreadfs/library_pthreadfs.js (hunk 2 above)
Comment (rstz) [reply pthreadfs/library_pthreadfs.js:3130]: Done.

File: pthreadfs/library_pthreadfs.js (hunk 3 above)
Comment (rstz) [reply pthreadfs/library_pthreadfs.js:3162]: Removed it.

**Comment by rstz** [comment]
Thank you for the review!

**Pull request merged**